### ISSUE TITLE
trying to compile in slackware 14.2 gives error

### ISSUE THREAD

**Issue by rspadim** [cminyard/gensio#21]
hi, i did an ./configure && make, and got this error, any idea?

```
# ./configure
checking for a BSD-compatible install... /usr/bin/ginstall -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /usr/x86_64-slackware-linux/bin/ld
checking if the linker (/usr/x86_64-slackware-linux/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/x86_64-slackware-linux/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... dlltool
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... mt
checking if mt is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/x86_64-slackware-linux/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking linux/version.h usability... yes
checking linux/version.h presence... yes
checking for linux/version.h... yes
checking for Linux epoll(7) interface with signals extension... yes
checking whether gcc is Clang... no
checking whether pthreads work with "-pthread" and "-lpthread"... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking OpenIPMI/ipmiif.h usability... no
checking OpenIPMI/ipmiif.h presence... no
checking for OpenIPMI/ipmiif.h... no
checking for pkg-config... /usr/bin/pkg-config
checking whether compiling and linking against OpenSSL works... yes
checking for library containing clock_gettime... none required
checking whether TIOCSRS485 is declared... yes
checking whether make supports nested variables... (cached) yes
checking for ANSI C header files... (cached) yes
checking for main in -lnsl... yes
checking for swig... /usr/bin/swig
checking SWIG version... 3.0.7
checking for SWIG library... /usr/share/swig/3.0.7
checking for python3... no
checking for python... /usr/bin/python
checking for a version of Python >= '2.1.0'... yes
checking for the distutils Python package... yes
checking for Python include path... -I/usr/include/python2.7
checking for Python library path... -L/usr/lib64 -lpython2.7
checking for Python site-packages path... /usr/lib64/python2.7/site-packages
checking python extra libraries... -lpthread -ldl  -lutil -lm
checking python extra linking flags... -Xlinker -export-dynamic
checking consistency of all components of python development environment... yes
checking for python version... 2.7.11
checking for python threads... yes
checking for sctp_bindx in -lsctp... no
checking for pam_start in -lpam... no
checking for getrandom... no
checking for ptsname_r... yes
checking for cfmakeraw... yes
checking for struct termios2.c_ispeed... yes
checking sys/un.h usability... yes
checking sys/un.h presence... yes
checking for sys/un.h... yes
checking for isatty... yes
checking for strcasecmp... yes
checking for strncasecmp... yes
checking for prctl... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating lib/libgensio.pc
config.status: creating lib/Makefile
config.status: creating include/Makefile
config.status: creating include/gensio/Makefile
config.status: creating include/gensio/gensio_version.h
config.status: creating tests/Makefile
config.status: creating tools/Makefile
config.status: creating swig/Makefile
config.status: creating swig/python/Makefile
config.status: creating man/Makefile
config.status: creating examples/Makefile
config.status: creating tests/runtest
config.status: creating tests/gensios_enabled.py
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
```
```
# make
make  all-recursive
make[1]: Entering directory '/root/gensio-master'
Making all in lib
make[2]: Entering directory '/root/gensio-master/lib'
  CC       gensio.lo
  CC       gensio_osops.lo
  CC       gensio_net.lo
  CC       gensio_udp.lo
  CC       gensio_stdio.lo
  CC       sergensio.lo
  CC       sergensio_telnet.lo
  CC       sergensio_serialdev.lo
  CC       uucplock.lo
  CC       gensio_selector.lo
  CC       gensio_ssl.lo
  CC       gensio_base.lo
  CC       gensio_filter_ssl.lo
  CC       gensio_filter_telnet.lo
  CC       telnet.lo
  CC       buffer.lo
  CC       gensio_ll_fd.lo
  CC       gensio_ll_gensio.lo
  CC       gensio_acc.lo
  CC       gensio_acc_gensio.lo
  CC       gensio_ll_ipmisol.lo
  CC       sergensio_ipmisol.lo
  CC       utils.lo
  CC       selector.lo
  CC       gensio_sctp.lo
  CC       gensio_filter_certauth.lo
  CC       gensio_certauth.lo
  CC       gensio_pty.lo
  CC       gensio_dummy.lo
  CC       gensio_echo.lo
  CC       gensio_mux.lo
  CC       gensio_file.lo
  CC       gensio_filter_msgdelim.lo
  CC       gensio_msgdelim.lo
  CC       gensio_filter_relpkt.lo
  CC       gensio_relpkt.lo
  CC       gensio_filter_trace.lo
  CC       gensio_trace.lo
  CC       gensio_filter_perf.lo
  CC       gensio_perf.lo
  CC       gensio_conacc.lo
  CCLD     libgensio.la
make[2]: Leaving directory '/root/gensio-master/lib'
Making all in include
make[2]: Entering directory '/root/gensio-master/include'
Making all in gensio
make[3]: Entering directory '/root/gensio-master/include/gensio'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/root/gensio-master/include/gensio'
make[3]: Entering directory '/root/gensio-master/include'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/root/gensio-master/include'
make[2]: Leaving directory '/root/gensio-master/include'
Making all in swig
make[2]: Entering directory '/root/gensio-master/swig'
Making all in .
make[3]: Entering directory '/root/gensio-master/swig'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/root/gensio-master/swig'
Making all in python
make[3]: Entering directory '/root/gensio-master/swig/python'
/usr/bin/swig -DHAVE_CONFIG_H -python  -o gensio_wrap.c \
        -I../../swig/python ../../swig/gensio.i
  CC       gensio_wrap.lo
  CCLD     _gensio.la
make[3]: Leaving directory '/root/gensio-master/swig/python'
make[2]: Leaving directory '/root/gensio-master/swig'
Making all in tests
make[2]: Entering directory '/root/gensio-master/tests'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/root/gensio-master/tests'
Making all in tools
make[2]: Entering directory '/root/gensio-master/tools'
  CC       ioinfo.o
  CC       ser_ioinfo.o
  CC       utils.o
  CC       localports.o
  AR       libgensiotool.a
  CC       gtlssh-shared.o
  AR       libgtlssh.a
  CC       gensiotool.o
gensiotool.c:95:13: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
     .seed = dummyrnd_seed,
             ^
gensiotool.c:95:13: note: (near initialization for 'dummyrnd.seed')
gensiotool.c:98:12: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
     .add = dummyrnd_add,
            ^
gensiotool.c:98:12: note: (near initialization for 'dummyrnd.add')
  CCLD     gensiot
  CC       gtlssh.o
gtlssh.c: In function 'check_cert_expiry':
gtlssh.c:305:9: warning: implicit declaration of function 'X509_get0_notAfter' [-Wimplicit-function-declaration]
     t = X509_get0_notAfter(x);
         ^
gtlssh.c:305:7: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
     t = X509_get0_notAfter(x);
       ^
  CCLD     gtlssh
gtlssh.o: In function `check_cert_expiry':
/root/gensio-master/tools/gtlssh.c:305: undefined reference to `X509_get0_notAfter'
collect2: error: ld returned 1 exit status
Makefile:576: recipe for target 'gtlssh' failed
make[2]: *** [gtlssh] Error 1
make[2]: Leaving directory '/root/gensio-master/tools'
Makefile:441: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/root/gensio-master'
Makefile:373: recipe for target 'all' failed
make: *** [all] Error 2
 
```

**Comment by rspadim** [comment]
* just to anyone else get here and see, without openssl works:

` ./configure --with-openssl=no`

**Comment by cminyard** [comment]
On Sun, Sep 13, 2020 at 07:02:49AM -0700, rspadim wrote:
> hi, i did an ./configure && make, and got this error, any idea?

This appears to be an openssl 1.0.x compatability issue.

Can you try the attached patch?

-corey

**Comment by rspadim** [comment]
hi @cminyard , i didn't found the patch in the message

**Comment by cminyard** [comment]
Well, apparently github here doesn't support .patch files.  I've appended .txt to the name, hopefully you can get thsi one.

[0001-tools-Fix-some-openssl-1.0-compatability-issues.patch.txt](https://github.com/cminyard/gensio/files/5219353/0001-tools-Fix-some-openssl-1.0-compatability-issues.patch.txt)

**Comment by cminyard** [comment]
Nothing heard, I'm assuming that patch fixed it.  It's in the latest release.
